### DISCUSSION TITLE
How do i inherit mapped_column or Column Class to add additional column properties ?

### DISCUSSION THREAD

**Discussion by saikiran2603** [sqlalchemy/sqlalchemy#11511] [Usage Questions] [answered]
I understand Column class has a property called "info" which can be used to store additional metadata about the column , but i am interested to extend the Column class possibly inherit it and add some additional properties to it, how do i go about it in 2.0. Since in 2.0 its recommended to used mapped_column which in turn calls a MappedColumn class . I see mapped class is using the Column class to assign properties here 

https://github.com/sqlalchemy/sqlalchemy/blob/5360cd115422d560b65456794cd700c7c87bfca6/lib/sqlalchemy/orm/properties.py#L585C1-L585C61

how do i add my custom Column class here , or is there any better way to extend the Column Class to add additional properties ?

**Comment by CaselIT** [comment]
Hi,

This use case it not supported, at least at the moment.
You can manually create the table and assign it to the orm class in the `__table__` attribute.

How so you intend to use the additional properties in an orm class? You are not usually using a table directly there. Do you have an example?

**Reply by saikiran2603** [reply]
Yes we wanted to store some properties like conditional formatting  , date format , precision etc which would be used on the user interface . Basically we wanted some attributes which are not related to database , but would be used in frontend for rendering the data fetched from the orm. What would be a better alternative to store additional attributes if extending the class is not supported ?

**Reply by CaselIT** [reply]
The info property is usually the suggested way of storing user data into sqlalchemy object.

**Reply by saikiran2603** [reply]
Can we override Column class and declare Table imperatively with ORM class , would that be supported ? 

for Eg:

```py
  class CustomColumn(Column):
  
      def __init__(self, *args: SchemaEventTarget, **dialect_kwargs: Any):
          # self.demo_parameter = demo_parameter
          print(dialect_kwargs)
          self.demo_parameter = dialect_kwargs['demo_parameter']
          dialect_kwargs.pop('demo_parameter')
          super().__init__(*args, **dialect_kwargs)

```

Something like this ? and the CustomColumn can be used to declare the ORM class imperatively

**Reply by CaselIT** [reply, marked as answer]
Sure, all previous orm mapping styles continue to be supported in v2.

The suggested is the one that uses Mapped and mapped_column because it's the one that best integrates with typing tools.

Note that I would instead use a function that abstracts how to get the custom props from a column or orm attribute. This way you could simply use a key in info

**Reply by saikiran2603** [reply]
Thanks for that @CaselIT  !
